### PR TITLE
open editors: use scoped context keys. Set them when context menu is triggered

### LINKED ISSUE #64735: Inconsistent behavior of context menu in Open editors view when it has regular and new (Untitled) files
Issue Type: <b>Bug</b>

VS Code version: Code - Insiders 1.30.0-insider (241923763dbeaf3b64443187e935d55b29f6125d, 2018-12-07T22:31:41.479Z)
OS version: Windows_NT x64 10.0.17763

When Open editors view has regular and new (not saved, with Untitled-X title) files - incorrect context menu might be rendered depending on order of files you call it for.

Steps to Reproduce:

1. Open some existing file. Right click on it in Open editor  will display **regular file context menu** (Similar options to menu in Explorer view - contains Open to the Side, Reveal in Explorer etc.)
1. Ctrl + N to create new file and type somethin in it.
1. Right-click on `Untiteled-1` in Open editors view. **New file context menu** (Save As..., Compare with selected etc.) will be shown.
1. Right-click on any previously opened existing file. **New file context menu** will be shown for existing file.
See `Context menu for noname-55.svg` below.

Next steps will allow to get regular context menu for new file (continue after steps above). I use this as workaround for issue reproduced above.

1. Right-click on any file in File explorer view.  **regular file context menu** is displayed.
1. Right-click on any `Untiteled-X` file in Open editors view. **Regular file context menu** is displayed. However actions like Open to the side and Open in Terminal doesn't do anything, while Select for compare, Compare with selected - works properly for unsaved files.
See `Context menu for Untiteled-3` below.

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 x 3408)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.88GB (17.43GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details>

Extensions: none

<details>
<summary>Context menu for Untiteled-3 file</summary>

image

</details>

<details>
<summary>Context menu for noname-55.svg</summary>

image

</details>

### PR DESCRIPTION
**Pull request by isidorn**
fixes #64735

@joaomoreno I would appreciate the review after what we have discussed.
@bpasero fyi on my creation and removal of the resourceContextKey

I suggest to not push this to recovery and just keep it in insiders.
In case we want to push it for recovery the open editors actions would require intense testing. I will comment more in the candidate issues.

### CODE REVIEW

**Comment by bpasero** [comment]
Looks a bit fishy to me and more like a hack. how does the explorer do it with files vs folders?

File: src/vs/workbench/parts/files/electron-browser/views/openEditorsView.ts [hunk 1]
```diff
@@ -390,13 +370,27 @@ export class OpenEditorsView extends ViewletPanel {
 		}
 
 		const element = e.element;
+		const scopedContextKeyService = this.contextKeyService.createScoped();
+		const groupFocusedContext = OpenEditorsGroupContext.bindTo(scopedContextKeyService);
+		const dirtyEditorFocused = DirtyEditorContext.bindTo(scopedContextKeyService);
+		const resourceContext = this.instantiationService.createInstance(ResourceContextKey, scopedContextKeyService);
+
+		if (element instanceof OpenEditor) {
```
Comment (umartinez22) [inline src/vs/workbench/parts/files/electron-browser/views/openEditorsView.ts]: Could add a commentary, about this logic?

**Comment by joaomoreno** [comment]
@bpasero Why does it feel like a hack?

When creating a context menu, we need the right context keys set in place such that the menu knows which actions to instantiate. This is much better than the alternative way of tracking the last focused list in a singleton and having the creator of the context menu access that singleton and ask "hey what was the last focused list, btw?" which also happens to rely on the fact selection comes before the context menu, as @isidorn mentioned. It seems logical to create the correct context, create a menu from it, instantiate actions from the menu and finally show the context menu. This avoids having a global singleton tracking generic list focus and existing just for the purposed of being asked _what was the last focused list_.

**Review by joaomoreno — APPROVED** [review]

**Comment by joaomoreno** [comment]
Closing as there are two ways these context keys are read:

- Keyboard shortcuts
- Context menus

**Pull request closed without merging**